pkg/network/server.go

```diff
@@ -264,6 +266,10 @@ func (s *Server) ID() uint32 {
 // Start will start the server and its underlying transport. Calling it twice
 // is an error.
 func (s *Server) Start() {
+	if !s.started.CompareAndSwap(false, true) {
```

Comment:
BTW, we need a test for this. But the test will work properly only after resolution of https://github.com/nspcc-dev/neo-go/pull/3307#discussion_r1492445285 since we need a reliable way to know if server was started.